cli/azd/pkg/infra/provisioning/custom/custom.go

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// Package custom contains a custom implementation of provider.Provider that no-ops the provisioning behavior
+// allowing people to use the hooks to implement custom behavior. This provider is registered for use when
+// this package is imported, and can be imported for side effects only to register the provider, e.g.:
+package custom
```

Comment:
I agree with @vhvb1989 here.  We shouldn't **need** to implement a custom provider for a noop / override scenario.  My thinking is more inline with @jongio 's approach to extending this behavior globally in hook configuration.